Src/Base/AMReX_ParallelContext.H

```diff
@@ -105,7 +105,11 @@ inline void set_last_frame_ofs (const std::string & filename) {
     frames.back().set_ofs_name(filename);
 }
 //! Note that it's the user's responsibility to free the MPI_Comm
-inline void pop () { frames.pop_back(); }
+inline void pop () {
+    if (frames.size() > 0u) {
```

Comment:
This might just hide errors somewhere else. One is not supposed to pop an empty container. For example, `std::vector::pop_back()` on empty container is undefined behavior.

Reply:
I saw this being called in particular in `ParallelContext::EndParallel()`:
https://github.com/AMReX-Codes/amrex/blob/ab99ea69089f4fffbfd727ed965d5ceb3d905baa/Src/Base/AMReX_ParallelDescriptor.cpp#L436